docker/Dockerfile.nvidia

```diff
@@ -114,74 +47,122 @@ ENV UCX_TLS=rc_x,sm,shm,cuda_copy,gdr_copy,cuda_ipc
+
+# Set base path based on version
+ENV HPCSDK_HOME=/opt/nvhpc
+ENV HPCSDK_CUPTI=/opt/CUPTI
+
+# required for nvidia-docker v1
+RUN echo "$HPCSDK_HOME/cuda/lib" >> /etc/ld.so.conf.d/nvidia.conf && \
+    echo "$HPCSDK_HOME/cuda/lib64" >> /etc/ld.so.conf.d/nvidia.conf && \
+    echo "$HPCSDK_HOME/compilers/lib" >> /etc/ld.so.conf.d/nvidia.conf && \
+    echo "$HPCSDK_HOME/comm_libs/mpi/lib" >> /etc/ld.so.conf.d/nvidia.conf && \
+    echo "$HPCSDK_CUPTI/lib64" >> /etc/ld.so.conf.d/nvidia.conf && \
+    echo "$HPCSDK_HOME/math_libs/lib64" >> /etc/ld.so.conf.d/nvidia.conf    
+    
+# Compiler, CUDA, and Library paths
+ENV CUDA_HOME $HPCSDK_HOME/cuda
+ENV CUDA_ROOT $HPCSDK_HOME/cuda/bin
+ENV PATH $HPCSDK_HOME/compilers/bin:$HPCSDK_HOME/cuda/bin:$HPCSDK_HOME/comm_libs/mpi/bin:${PATH}
+ENV LD_LIBRARY_PATH $HPCSDK_HOME/cuda/lib:$HPCSDK_HOME/cuda/lib64:$HPCSDK_HOME/compilers/lib:$HPCSDK_HOME/math_libs/lib64:$HPCSDK_HOME/comm_libs/mpi/lib:$HPCSDK_CUPTI/lib64:bitcomp_DIR:${LD_LIBRARY_PATH}
 ENV CPATH $HPCSDK_HOME/comm_libs/mpi/include:${CPATH}
 
-#MPI 3
-# Do Nothing
+# MPI 3
 RUN if [ "x$MPI_VER" = "x4" ]; then \
         rm -f  $HPCSDK_HOME/comm_libs/mpi && \
         ln -sf $HPCSDK_HOME/comm_libs/openmpi4/openmpi-4.0.5 \
                $HPCSDK_HOME/comm_libs/mpi ; \
     fi;  \
-    if [ "x$MPI_VER" = "HPCX" ]; then \
+    if [ "$MPI_VER" = "HPCX" ]; then \
         rm -f  $HPCSDK_HOME/comm_libs/mpi && \
         ln -sf $HPCSDK_HOME/comm_libs/hpcx/latest/ompi \
                $HPCSDK_HOME/comm_libs/mpi ; \
     fi;
 
-# Install pip dependencies and devito as a pip package
+# Install python nvidia dependencies
 RUN python3 -m venv /venv && \
     /venv/bin/pip install --no-cache-dir --upgrade pip && \
-    /venv/bin/pip install --no-cache-dir wheel && \
-    /venv/bin/pip install --no-cache-dir -e /app/devito[extras,nvidia] && \
-    # Need extra CFLAGS for mpi4py
-    CFLAGS=-noswitcherror /venv/bin/pip install --no-cache-dir -r /app/devito/requirements-mpi.txt && \
+    /venv/bin/pip install --no-cache-dir -r https://raw.githubusercontent.com/devitocodes/devito/master/requirements-nvidia.txt && \
+    # Install jupyter and setup nvidia configs.
+    /venv/bin/pip install --no-cache-dir jupyter && \
+    /venv/bin/jupyter serverextension enable dask_labextension && \
     rm -rf ~/.cache/pip
 
-ADD docker/run-jupyterlab.sh /jupyter
-ADD docker/run-tests.sh /tests
-ADD docker/run-print-defaults.sh /print-defaults
-ADD docker/entrypoint.sh /docker-entrypoint.sh
+########################################################################
+# NVC config
+########################################################################
+FROM sdk-base as nvc
+
+# Make devito env vars file and extras
 ADD docker/nvdashboard.json /app/nvdashboard.json
 
-RUN chmod +x /print-defaults /jupyter /tests /docker-entrypoint.sh && \
-    /venv/bin/jupyter serverextension enable dask_labextension 
-#    /venv/bin/jupyter lab workspaces import /app/nvdashboard.json
+# mpi4py
```

Comment:
I'm a bit baffled we don't set CC=nvc (while MPICC=mpicc should be unnecessary since mpicc in PATH is nvidia's mpicc). We used to in our wiki installation instructions

Reply:
>I'm a bit baffled we don't set CC=nvc

It's not actually compiling openmpi so you don't need CC